Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/PerfLib.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib
+{
+    using System;
+    using System.Globalization;
+
+    internal class PerfLib : IQuickPulsePerfLib
+    {
+        private static PerfLib library = null;
+
+        private PerformanceMonitor performanceMonitor;
+
+        private PerfLib()
+        {
+        }
+
+        public static PerfLib GetPerfLib()
+        {
+            library = library ?? new PerfLib();
+
+            return library;
+        }
+
+        public CategorySample GetCategorySample(int categoryIndex, int counterIndex)
+        {
+            byte[] dataRef = this.GetPerformanceData(categoryIndex.ToString(CultureInfo.InvariantCulture));
+            if (dataRef == null)
+            {
+                throw new InvalidOperationException("Could not read data for category index " + categoryIndex);
+            }
+
+            return new CategorySample(dataRef, categoryIndex, counterIndex, this);
+        }
+
+        public void Initialize()
+        {
+            this.performanceMonitor = new PerformanceMonitor();
+        }
+
+        public void Close()
+        {
+            this.performanceMonitor?.Close();
+
+            library = null;
```

Comment:
So it looks like Dispose, but it isn't, and the library can be recreated in GetPerfLib, I realize you copied this from the CLR but it's definitely not the usual pattern.  This also doesn't look thread safe if closing the performanceMonitor will cause issues for another thread trying to still use the same instance.